kili/services/asset_import/legacy.py

```diff
@@ -0,0 +1,375 @@
+# pylint: skip-file
+"""Legacy code for importing assets, to be removed soon"""
+
+import csv
+import mimetypes
+import os
+from json import dumps
+from typing import List, Union
+from uuid import uuid4
+
+from kili.authentication import KiliAuth
+from kili.graphql.operations.asset.mutations import (
+    GQL_APPEND_MANY_FRAMES_TO_DATASET,
+    GQL_APPEND_MANY_TO_DATASET,
+)
+from kili.helpers import (
+    check_file_mime_type,
+    encode_base64,
+    format_result,
+    get_data_type,
+    is_url,
+)
+from kili.orm import Asset
+from kili.utils import bucket, pagination
+
+from .types import AssetToImport
+
+
+class LegacyImporter:
+    """Legacy Asset importer
+    First switch form new list of dict paradigm to the old set of list one, then import assets"""
+
+    def __init__(
+        self,
+        auth: KiliAuth,
+        project_id: str,
+        input_type: str,
+        assets: List[AssetToImport],
+    ):
+        self.input_type = input_type
+        self.project_id = project_id
+        self.auth = auth
+        self.content_array = assets[0].get("content", None) and [
```

Comment:
it is a better practice to remove the arrays from the attributes, and make `assets` an argument of the `.import_assets` method: `def import(assets: List[AssetToImport])`